extensions/metadata-3.3.md

```diff
@@ -0,0 +1,389 @@
+with a `ERR_KEYINVALID` numeric.
+
+Servers MAY additionally respond to requests to subscribe to a key that the
+client has no privilege to access with a `ERR_KEYNOPERMISSION` numeric.
+Even if a server does that, the subscription MUST still be successful.
+This is because in this case the `ERR_KEYNOPERMISSION` numeric only serves as a
+warning, indicating that the client will not receive METADATA events about this
+key unless it acquires the necessary (implementation defined) privileges later.
+
+### METADATA UNSUB
+
+This subcommand is used to unsubscribe from metadata keys.
+
+Syntax:
+
+    METADATA * UNSUB <key1> [<key2> ...]
+
+The reply of the server MUST be zero or more numerics of the following types
+in any order: `RPL_METADATAUNSUBOK`, `ERR_KEYINVALID`.
+Then the server MUST end the reply with one `RPL_METADATAEND` numeric.
+
+If a client successfully unsubscribes from a key it is subscribed to then the
+key MUST appear in a `RPL_METADATAUNSUBOK` reply numeric.
+
+If a client tries to unsubscribe from a key that it is not subscribed to then
+the client remains not subscribed to the key and the key MUST appear at least
+once in a RPL_METADATAUNSUBOK numeric in the reply.
+
+Servers MUST respond to requests to subscribe to a key whose name is invalid
+with a `ERR_KEYINVALID` numeric.
+
+### METADATA SUBS
+
+This subcommand can be used to get a list of keys which the client is
+subscribed to.
+
+Syntax:
+
+    METADATA * SUBS
+
+The server MUST reply with zero or more `RPL_METADATASUBS` numerics and then
+one `RPL_METADATAEND` numeric.
+
+The replied `RPL_METADATASUBS` numerics, collectively, MUST contain all keys
+the client is subscribed to exactly once and MUST NOT contain keys the client
+is not subscribed to.
+
+## Numerics
+
+| No. | Label                       | Format                            |
+| --- | --------------------------- | ----------------------------------|
+| 770 | `RPL_METADATASUBOK`         | `:<Key1> [<Key2> ...]`            |
+| 771 | `RPL_METADATAUNSUBOK`       | `:<Key1> [<Key2> ...]`            |
+| 772 | `RPL_METADATASUBS`          | `:<Key1> [<Key2> ...]`            |
+| 773 | `ERR_METADATATOOMANYSUBS`   | `<Key>`                           |
+
+### RPL_METADATASUBS
+
+The order of the keys is undefined.
+
+### ERR_METADATATOOMANYSUBS
+
+The <key> parameter of this numeric is the first key that the client was not
+subscribed to.
+
+## Compatibility with `metadata-notify`
+
+It is pointless to use `metadata-notify-2` and `metadata-notify` (as described
+in the [metadata v3.2 specification](http://ircv3.net/specs/core/metadata-3.2.html))
+together.
+
+This is because `metadata-notify` implicitly subscribes the client to all keys,
+while `metadata-notify-2` requires the client to tell the keys it wants to
+subscribe to.
+
+Servers and clients MAY support both of the mentioned extensions, but MUST NOT
+negotiate both of them at the same time in the same connection.
```

Comment:
Any good reason to do it that way? I've deliberately tried to avoid adding special cases like "if both are present, accept only one" or "if the order is X Y accept if Y X then reject". Servers must accept all caps in a CAP REQ or reject them all per the cap spec.